### ISSUE TITLE
Inconsistent session data access

### ISSUE THREAD

**Issue by IvanLisz** [nksaraf/vinxi#120]
1. Session cookie is always accessible via event.request.headers.get('cookie'), but data is not event.context.sessions (used in useSession).

2. Sometimes the session data is there, and sometimes it's undefined.
Using a middleware to check sessions (just with an await useSession(event, sessionConfig)) makes it work the route level.

____________

middleware:
```
import { createMiddleware, useSession } from "@solidjs/start/server";
import { sessionConfig } from "~/lib/session";

export default createMiddleware({
  onRequest: [
    async (event) => {
        await useSession(event, sessionConfig)
    },
  ],
});
```

route:
```
import { cache, createAsync, redirect, type RouteSectionProps } from "@solidjs/router";
import { Show } from "solid-js";
import { getRequestEvent } from "solid-js/web";
import { sessionConfig } from "~/lib/session";
import { useSession } from 'vinxi/server'

const getUser = cache(async () => {
  "use server";

  let token;
  try {

    const event = getRequestEvent()!;
    const session = await useSession(event, sessionConfig)
  
    token = session.data.token;
    console.log(Date.now(), event.request.headers.get('cookie')?.length, token, event.context.sessions)
  } catch (e){
    console.log(e);
  }

  if(!token) {
    console.log('redirect')
    throw redirect('/login')
  }
  
  return token;  
}, "token");

export const route = {
  load: getUser
};

export default function AuthenthicatedRoutes(props: RouteSectionProps) {
  const token = createAsync(getUser, { deferStream: true });

  console.log('--------------')

  return <Show when={token()}>{props.children}</Show>;
}
```

**Comment by nksaraf** [comment]
I think this should be fixed by #121. I wonder if you have a way to try it and confirm that it works. Or @edivados should we add this case in our test so we can confirm its fixed.

**Comment by IvanLisz** [comment]
lgtm 🚀
